cube/cube.c

```diff
@@ -4281,10 +4639,85 @@ static void demo_init(struct demo *demo, int argc, char **argv) {
             demo->force_errors = true;
             continue;
         }
-
+        if ((strcmp(argv[i], "--wsi") == 0) && (i < argc - 1)) {
+            size_t argc_len = strlen(argv[i + 1]);
+            for (size_t argc_i = 0; argc_i < argc_len; argc_i++) {
+                argv[i + 1][argc_i] = toupper(argv[i + 1][argc_i]);
+            }
+            WSI_PLATFORM selection = wsi_from_string(argv[i + 1]);
+            if (selection == WSI_PLATFORM_INVALID) {
+                printf(
+                    "The --wsi parameter %s is not a supported WSI platform. The list of available platforms is available from "
+                    "--help\n",
+                    (const char *)&(argv[i + 1][0]));
+                fflush(stdout);
+                exit(1);
+            }
+            demo->wsi_platform = selection;
+            i++;
+            continue;
+        }
 #if defined(ANDROID)
         ERR_EXIT("Usage: vkcube [--validate]\n", "Usage");
 #else
+
+        // Making the help for --wsi nice requires a little extra work since the list depends on what is available at
+        // compile time
+        size_t max_str_len = 100;
+        char *available_wsi_platforms = (char *)malloc(max_str_len);
+        memset(available_wsi_platforms, 0, max_str_len);
+#if defined(VK_USE_PLATFORM_XCB_KHR)
+        strncat(available_wsi_platforms, "XCB", max_str_len);
```

Comment:
It is a bit inconsistent that in `vkcube` the options are lowercase in the help and in `vkcubepp` the options are uppercase. It does seem to work in both ways in either though. But can the help be made consistent to avoid confusion?

```
$ ./vkcube --help
	[--wsi <XCB|XLIB|WAYLAND|DISPLAY>]
$ ./vkcubepp --help
	[--wsi <xcb|xlib|wayland|display>]
```

Reply:
Yes, that was a bit of a "I had to rewrite the code to use C++ and didn't notice that they were different". I went with lowercase since its simpler to type out. I kept the "tolower" transformation in case someone types it in all caps, since it doesn't hurt to do that for a user.